recipe/build.sh

```diff
@@ -1,19 +1,23 @@
+#!/usr/bin/env bash
+set -eux
+
+sed -i -e \
```

Comment:
For context, this is modifying this line: https://github.com/hunspell/hunspell/blob/019e9769b655c0eaedfdfafefac81821cf4d1b7e/src/tools/hunspell.cxx#L118-L122

Is it not possible to do this with a patch?

Reply:
Yeah, I dunno: i've seen patterns where there are patches at rest that then get `sed`ed, but I assume the prefix-rewriting stuff needs actual code up in there.

Reply:
I couldn't find anything in the docs about a way to replace the prefix in a patch, but maybe someone knows how to do it? Otherwise I guess this sed line is fine.

Reply:
(deferring to [runtime `getenv("CONDA_PREFIX")`](https://github.com/hunspell/hunspell/blob/019e9769b655c0eaedfdfafefac81821cf4d1b7e/src/tools/hunspell.cxx#L118-L122) would work, but seems less palatable... and has given me trouble on other feedstocks)

Reply:
Actually maybe it would be better to use `patch` with an inline heredoc instead of `sed`. The advantage of patch is 1. it's more self documenting what it is changing, and 2. it will properly fail if the upstream source changes, and it also won't accidentally apply to the wrong line (e.g., it probably doesn't matter, butt there is another `#define LIBDIR` line in that file https://github.com/hunspell/hunspell/blob/019e9769b655c0eaedfdfafefac81821cf4d1b7e/src/tools/hunspell.cxx#L73). 

The upstream changes part is the thing that particularlly worries me. For instance, the line in the github source has a trailing `\`. Is that handled by this sed replacement?

Reply:
if you have a vision, please feel free to push!

Reply:
i guess a more precise alternative would be:

```bash
"s;/usr/share/hunspell:;${PREFIX}/share/hunspell:${PREFIX}/share/hunspell_dictionaries:/usr/share/hunspell:"
```

Reply:
I pushed up the idea of what I'm talking about. I'm not sure if I need to be passing some more command line arguments to `patch`.

Reply:
Here's the [build that failed](
https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=1148838&view=logs&j=d6b58996-039f-5e48-56bf-c3a016e5cd7f&t=dd6a23d9-2356-5264-64bf-579875a5d8d6&l=1427) with:

```
$SRC_DIR/conda_build.sh: line 26: CROSSCOMPILING_EMULATOR: unbound variable
```

Reply:
> passing some more command line arguments to

Yeah, I don't know... quoting/-heredoc with expansion always gives me a headache.

I tried the thing [alluded to above](https://github.com/conda-forge/hunspell-feedstock/pull/13#discussion_r1917468251) with a `.patch` on disk, and then `sed` patterns we _know_ won't change without us knowing... :crossed_fingers:

Reply:
Huzzah, back to :heavy_check_mark:.